src/utils/linked_list.rs

```diff
@@ -0,0 +1,383 @@
+    pub fn new() -> Self {
+        Self { first: None }
+    }
+
+    /// Add an item at the head of the list
+    ///
+    /// An item must be mutable to be added to a list.
+    ///
+    /// When an item is added, the [`LinkedList`] mutably borrows the item for as long as the item
+    /// is in the list. You can't use mutable or immutable reference to the item while it is added
+    /// to the list (because of this exclusive borrow). To represent the added item while
+    /// interacting with the [`LinkedList`] you must use other object, which created when you add
+    /// an item: a [token](ListItemToken). When an item is added to the list, the list exclusively
+    /// borrows the item from you, however in exchange you get a [token](ListItemToken), like in a
+    /// cloakroom. You can use the token to get the exclusively borrowed item back.
+    ///
+    /// # Examples
+    /// ```
+    /// use nrf_radio::utils::linked_list::{LinkedList, ListItem};
+    ///
+    /// let mut list = LinkedList::new();
+    /// let mut item = ListItem::new("Hello");
+    ///
+    /// let token = list.push(&mut item);
+    /// ```
+    pub fn push(&mut self, item: &'list mut ListItem<'list, T>) -> ListItemToken<'list, T> {
+        if item.next.is_some() {
+            panic!("Next link of an item is set indicating it is added to a list. But if it is added to a list it's user shall not have access to mutable reference to an item to call this function.");
+        }
+
+        let token = ListItemToken { ptr: item };
+
+        let next = self.first.take();
+        item.next = next;
+        self.first.replace(item);
+
+        token
+    }
+
+    /// Get an item from the head of the list
+    ///
+    /// This action removes the retrieved item from the list.
+    /// Actually, you don't get an owenrship of the item, but you exclusively borrows the item from
+    /// its owner (the borrow is passed from the list to you).
+    ///
+    /// You don't need a [token](ListItemToken) to pop an item from the list. Tokens are needed to
+    /// precisely identify one item, but popping gets any item that happens to be at the head.
+    ///
+    /// # Examples
+    /// ```
+    /// use nrf_radio::utils::linked_list::{LinkedList, ListItem};
+    ///
+    /// let mut list = LinkedList::new();
+    /// let mut item = ListItem::new("Hello");
+    ///
+    /// let token = list.push(&mut item);
+    ///
+    /// let item_option = list.pop();
+    /// assert!(item_option.is_some());
+    /// let item_ref = item_option.unwrap();
+    /// assert_eq!(**item_ref, "Hello");
+    ///
+    /// let item_option = list.pop();
+    /// assert!(item_option.is_none());
+    /// ```
+    pub fn pop(&mut self) -> Option<&'list mut ListItem<'list, T>> {
+        let first = self.first.take();
+        if let Some(first) = first {
+            let next = first.next.take();
+            self.first = next;
+            Some(first)
+        } else {
+            None
+        }
+    }
```

Comment:
```suggestion
    pub fn pop(&mut self) -> Option<&'list mut ListItem<'list, T>> {
        let item = self.first.take();
        
        if let Some(item) = item {
            self.first = item.next.take();
        }
        
        item
    }
```

Reply:
Does not work. `item` is moved (or partially moved) to `if let` block in line 169, so it cannot be used after the block in line 173. However it can be returned from within the block like this:

```
        let item = self.first.take();

        if let Some(item) = item {
            self.first = item.next.take();
            return Some(item)
        } else {
            None
        }
```

However it's similar to original solution, so I'm not sure if it makes sense changing

Reply:
Ah sorry,
then it will be
```rust
    pub fn pop(&mut self) -> Option<&'list mut ListItem<'list, T>> {
        let mut item = self.first.take();

        if let Some(item) = item.as_mut() {
            self.first = item.next.take();

        }

        item
    }
```

Reply:
Right, done